StreamReader.h

```diff
@@ -40,6 +40,7 @@ class FileStreamReader : public SeekableStreamReader {
 
 private:
 	std::ifstream file;
+	uint64_t streamSize;
```

Comment:
This makes me a bit squeamish. We can't know for certain the stream size won't change due to some external process. I'm kind of thinking of reverting the recent size checks on the file stream, documenting the read past end of file behaviour as undefined for now, and opening a new issue to track this as a future problem.